client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -1573,12 +1577,32 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
         latestPartitionLocation.remove(shuffleId)
         commitManager.removeExpiredShuffle(shuffleId)
         changePartitionManager.removeExpiredShuffle(shuffleId)
-        val unregisterShuffleResponse = requestMasterUnregisterShuffle(
-          UnregisterShuffle(appUniqueId, shuffleId, MasterClient.genRequestId()))
-        // if unregister shuffle not success, wait next turn
-        if (StatusCode.SUCCESS == Utils.toStatusCode(unregisterShuffleResponse.getStatus)) {
+        if (!batchRemoveExpiredShuffles) {
+          val unregisterShuffleResponse = requestMasterUnregisterShuffle(
+            UnregisterShuffle(appUniqueId, shuffleId, MasterClient.genRequestId()))
+          // if unregister shuffle not success, wait next turn
+          if (StatusCode.SUCCESS == Utils.toStatusCode(unregisterShuffleResponse.getStatus)) {
+            unregisterShuffleTime.remove(shuffleId)
+          }
+        } else {
+          batchRemoveShuffleIds.add(shuffleId)
+        }
+      }
+    }
+    if (!batchRemoveShuffleIds.isEmpty) {
```

Comment:
```suggestion
    if (batchRemoveExpiredShuffles) {
```
Use the `batchRemoveExpiredShuffles` for condition would be more easy to handle and understand. `batchRemoveShuffleIds` are only collected when `batchRemoveExpiredShuffles` is set to be `true` anyway.

Reply:
It is not necessary that there are shuffle id's to remove.
So the `!batchRemoveShuffleIds.isEmpty` is required here to prevent sending `BatchUnregisterShuffles` with no shuffle ids.

Reply:
OK. But to be honest, the current implementation is far from the best.

Reply:
I am not sure what you mean - if there are suggestions to improve it in a PR, that would be great.